tracer/test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -426,18 +447,52 @@ public string GetTargetFramework()
             return $"net{_major}{_minor}{_patch ?? string.Empty}";
         }
 
+        public void EnabledWindowsNamedPipes(string tracePipeName = null, string statsPipeName = null)
+        {
+            TransportType = TestTransports.WindowsNamedPipe;
+            throw new NotImplementedException("The MockTracerAgent does not yet support Windows Named Pipes");
+        }
+
+        public void EnableUnixDomainSockets()
+        {
+#if NETCOREAPP3_1_OR_GREATER
+            TransportType = TestTransports.Uds;
+#else
+            // Unsupported
+            TransportType = TestTransports.Tcp;
+#endif
+        }
+
         public MockTracerAgent GetMockAgent(bool useStatsD = false)
         {
-            // Strategy pattern for agent transports goes here
-            var agentPort = TcpPortProvider.GetOpenPort();
-            var agent = new MockTracerAgent(agentPort, useStatsd: useStatsD);
+            MockTracerAgent agent = null;
 
-            _output.WriteLine($"Assigned port {agent.Port} for the agentPort.");
+            EnableUnixDomainSockets();
```

Comment:
To be removed once we're confident UDS causes no failures in the whole test suite.